lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -51,67 +56,77 @@ public function __invoke(AssetUpdateTasksMessage $message): void
+            $save = true;
+            $saveParams['versionNote'] = 'PDF scan result';
         }
 
-        $savedNeeded = false;
-        $pageCount = $asset->getCustomSetting('document_page_count');
-        if (!$pageCount || $pageCount === 'failed') {
-            $savedNeeded |= $asset->processPageCount();
-        }
+        if ($asset->isPageCountProcessingEnabled()) {
+            $pageCount = $asset->getCustomSetting('document_page_count');
+            if (!$pageCount || $pageCount === 'failed') {
+                if (!$asset->processPageCount() || $asset->getCustomSetting('document_page_count') === 'failed') {
+                    $asset->setCustomSetting(Asset::CUSTOM_SETTING_PROCESSING_FAILED, true);
+                    $this->logger->warning(sprintf('Failed processing page count for document asset %s.', $asset->getId()));
+                }
 
-        if ($savedNeeded) {
-            $this->saveAsset($asset);
+                $save = true;
+            }
         }
 
-        if ($asset->getCustomSetting('document_page_count') !== 'failed') {
+        if ($asset->isThumbnailsEnabled() && !$asset->getCustomSetting(Asset::CUSTOM_SETTING_PROCESSING_FAILED)) {
             $asset->getImageThumbnail(Asset\Image\Thumbnail\Config::getPreviewConfig())->generate(false);
-        } elseif ($savedNeeded) {
-            $this->logger->warning(sprintf('Failed processing page count for document asset %s.', $asset->getId()));
+        }
+
+        if ($save) {
+            $this->saveAsset($asset, $saveParams);
         }
     }
 
     private function processVideo(Asset\Video $asset): void
     {
+        $asset->removeCustomSetting(Asset::CUSTOM_SETTING_PROCESSING_FAILED);
+        $failed = true;
+
         if ($duration = $asset->getDurationFromBackend()) {
             $asset->setCustomSetting('duration', $duration);
-        } else {
-            $asset->removeCustomSetting('duration');
+            if ($dimensions = $asset->getDimensionsFromBackend()) {
+                $asset->setCustomSetting('videoWidth', $dimensions['width']);
+                $asset->setCustomSetting('videoHeight', $dimensions['height']);
+                $failed = false;
+            }
         }
 
-        if ($dimensions = $asset->getDimensionsFromBackend()) {
-            $asset->setCustomSetting('videoWidth', $dimensions['width']);
-            $asset->setCustomSetting('videoHeight', $dimensions['height']);
-        } else {
+        if ($failed) {
+            $asset->setCustomSetting(Asset::CUSTOM_SETTING_PROCESSING_FAILED, true);
+            $asset->removeCustomSetting('duration');
             $asset->removeCustomSetting('videoWidth');
             $asset->removeCustomSetting('videoHeight');
         }
 
         $sphericalMetaData = $asset->getSphericalMetaDataFromBackend();
-        if (!empty($sphericalMetaData)) {
+        if ($sphericalMetaData !== $asset->getCustomSetting('SphericalMetaData')) {
```

Comment:
AFAIK `addToUpdateTaskQueue` is after `getDataChanged` condition, but not sure in case the same file is uplodaded again and has no effectively changes, would still triggers DataChanged (or eg. by copy/paste in the object tree?), the `SsphericalMetaData` might be the same.

Also if former data was valid, but new one is nullish, then should remove the customSetting as it was doing before, something more like

```suggestion
        if (!sphericalMetaData) {
            $asset->removeCustomSetting('SphericalMetaData');
        } elseif ($sphericalMetaData !== $asset->getCustomSetting('SphericalMetaData')) {
            $asset->setCustomSetting('SphericalMetaData', $sphericalMetaData);
        }
```

Reply:
Fixed.